crates/admin/protobuf/cluster_ctrl_svc.proto

```diff
@@ -90,12 +90,19 @@ message CreatePartitionSnapshotRequest { uint32 partition_id = 1; }
 message CreatePartitionSnapshotResponse { string snapshot_id = 1; }
 
 message ChainExtension {
+  reserved 5;
```

Comment:
TIL you can [do this](https://protobuf.dev/best-practices/dos-donts/) in proto!

Reply:
It's not going to be a great experience if someone mixes different versions of restate-server and restatectl. Not fatal in this case but you might be wondering why you're getting validation errors on a seemingly legit reconfigure request. At some point we should start including explicit version info in these messages.

Should we keep the field on the server and explicitly reject the request as a validation error if `params` is used for now?

Reply:
I get your concern. I definitely missed that one. My focus was to make sure it doesn't do harm if `params is missing and the system will just reconfigure based on the cluster configuration.

I like your idea. I will put this field back and just fail if it's set